### ISSUE TITLE
Add numerical correctness tests for more ops

### ISSUE THREAD

**Issue by m96-chan** [m96-chan/NxPU#101]
Currently numerical correctness tests only cover matmul, vecadd, vecsub, vecmul, and relu. Extend to cover Conv2D, pooling, reduce ops, softmax, batch normalization, transpose, and attention patterns.
